### PR TITLE
Smart Search - move Filters - v2 to correct merge conflict

### PR DESCRIPTION
**Pull request by pe7er**
New PR to solve Merge Conflict of PR #6959 "Smart Search - move Filters (3x) from left to [Search Tools] "

Please follow the testing procedure of PR #6959

---

This PR moves the Filter option of all three views of **Smart Search** to the [Search Tools] button.
(See Issue #6941 regarding the inconsistency with Filter options).
## Test instructions

You have to enable the **Content - Smart Search** Plugin, and **Index your test website**
Please check the 3 different views before & after the patch: 
test all the filters and search box, clear results, choose max amount of results on a page.
### Check the current position of the Filters

In back-end > Components > Smart Search > Indexed Content
The Filters are on the left hand side

screen shot 2015-05-15 at 14 22 52
### This PR changes moves the Filter

from the left to the middle column and will be visible when you click on the [Search Tools] button.

screen shot 2015-05-15 at 14 22 52
### Check the current position of the Filters

In back-end > Components > Smart Search > Content Maps
The Filters are on the left hand side

screen shot 2015-05-15 at 14 22 52
### This PR changes moves the Filter

from the left to the middle column and will be visible when you click on the [Search Tools] button.

screen shot 2015-05-15 at 14 22 52
### Check the current position of the Filters

In back-end > Components > Smart Search > Search Filters
The Filters are on the left hand side

screen shot 2015-05-15 at 14 22 52
### This PR changes moves the Filter

from the left to the middle column and will be visible when you click on the [Search Tools] button.

screen shot 2015-05-15 at 14 22 52
## Note

To Clear the Filters / Search box, you can use the [Clear] button.
However, sometimes you have to press 2 times to clear the results.
I don't know the cause and how to solve it...

### CODE REVIEW

**Comment by jduerscheid** [comment]
I have tested this item :white_check_mark: successfully on 9a763fbba7551d868abb60e11a66faac3942f750<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8140">issues.joomla.org/joomla-cms/8140</a>.</sub>

File: administrator/components/com_finder/views/index/tmpl/default.php [hunk 1]
```diff
@@ -60,19 +60,7 @@
...
-				<button type="submit" class="btn hasTooltip" title="<?php echo JHtml::tooltipText('JSEARCH_FILTER_SUBMIT'); ?>"><span class="icon-search"></span></button>
-				<button type="button" class="btn hasTooltip" title="<?php echo JHtml::tooltipText('JSEARCH_FILTER_CLEAR'); ?>" onclick="document.getElementById('filter_search').value='';this.form.submit();"><span class="icon-remove"></span></button>
-			</div>
-			<div class="btn-group pull-right hidden-phone">
-				<label for="limit" class="element-invisible"><?php echo JText::_('JFIELD_PLG_SEARCH_SEARCHLIMIT_DESC');?></label>
-				<?php echo $this->pagination->getLimitBox(); ?>
-			</div>
-		</div>
+		<?php echo JLayoutHelper::render('joomla.searchtools.default', array('view' => $this)); ?>
 		<div class="clearfix"> </div>
```
Comment (roland-d) [inline administrator/components/com_finder/views/index/tmpl/default.php]: @pe7er I think the clearfix should be removed here. You did that in the old PR. Can you check please?

File: administrator/components/com_finder/views/maps/tmpl/default.php [hunk 2]
```diff
@@ -46,15 +46,7 @@
...
-			<div class="filter-search btn-group pull-left">
-				<input type="text" name="filter_search" id="filter_search" placeholder="<?php echo JText::_('JSEARCH_FILTER'); ?>" value="<?php echo $this->escape($this->state->get('filter.search')); ?>" class="hasTooltip" title="<?php echo JHtml::tooltipText('COM_FINDER_FILTER_SEARCH_DESCRIPTION'); ?>" />
-			</div>
-			<div class="btn-group pull-left">
-				<button type="submit" class="btn hasTooltip" title="<?php echo JHtml::tooltipText('JSEARCH_FILTER_SUBMIT'); ?>"><span class="icon-search"></span></button>
-				<button type="button" class="btn hasTooltip" title="<?php echo JHtml::tooltipText('JSEARCH_FILTER_CLEAR'); ?>" onclick="document.getElementById('filter_search').value='';this.form.submit();"><span class="icon-remove"></span></button>
-			</div>
-		</div>
+		<?php echo JLayoutHelper::render('joomla.searchtools.default', array('view' => $this)); ?>
 		<div class="clearfix"> </div>
```
Comment (roland-d) [inline administrator/components/com_finder/views/maps/tmpl/default.php]: @pe7er The clearfix should be mroeved I think as you did in your old PR. Can you check please?

File: administrator/components/com_finder/views/maps/view.html.php [hunk 3]
```diff
@@ -92,20 +94,5 @@ protected function addToolbar()
 			JToolbarHelper::deleteList('', 'maps.delete');
 			JToolbarHelper::divider();
 		}
-
-		JHtmlSidebar::setAction('index.php?option=com_finder&view=maps');
```
Comment (roland-d) [inline administrator/components/com_finder/views/maps/view.html.php]: @pe7er In your old PR this line is still there, should it remain or be removed?

**Comment by pe7er** [comment]
@roland-d I think that JHtmlSidebar::setAction can be removed.
I've removed the clearfix in all three list views.

**Comment by n9iels** [comment]
I applied the patch before activating the content plugin. The message that tells me that is now very close to the search filters. Maybe we can add some padding there?
smart search message

I also noticed that the filter on "Content maps" are always visible, and for some reason the filter on the right is always marked as an active filter (There are blue lines around it).
On other places where we use this filters (com_content for example) the blue line only appears when we choose an option of the filter.

**Comment by pe7er** [comment]
Thanks @n9iels, I corrected the Branch filter on "Content maps".
Now it is not activated automatically anymore.

btw: I did not change the padding of the warning message...

**Comment by n9iels** [comment]
Yes, that works for me! :smile:  I see now the same problem appears for "indexed content", the search filter "Select type of content" is always active over there.

Also the search filter on "Content mapping" are always visible on that page. Or is that done by a certain reason?

**Comment by n9iels** [comment]
About that padding and the message. Strange enough the message appears below the search filters. Other messages like "save" and "delete" appear above the messages.

Seems like all other message than "successfully saved", "successfully deleted" etc. appear below the filters. Not a problem for this PR I think so.

**Comment by pe7er** [comment]
Thanks for testing! 

No, I did not set the filter to "always active" on purpose. I'll check if I can correct it.

If someone knows the cause of the problem, please correct it with a PR, 
or tell me what to change & I'll do so...

**Comment by roland-d** [comment]
@pe7er The "always active" happens because you have set a default value of 0 here: https://github.com/joomla/joomla-cms/pull/8140/files#diff-762a4c09ff9b1a26880f1391232575c2R23

The first option you give to the dropdown has a value of 0:
https://github.com/joomla/joomla-cms/pull/8140/files#diff-762a4c09ff9b1a26880f1391232575c2R29

that is why the first entry becomes an active one. I guess you can drop the default value setting.

**Comment by zero-24** [comment]
@pe7er Travis cry

```
FILE: ...d/joomla/joomla-cms/administrator/components/com_finder/models/maps.php
--------------------------------------------------------------------------------
FOUND 2 ERROR(S) AFFECTING 2 LINE(S)
--------------------------------------------------------------------------------
 176 | ERROR | Expected "if (...)\n...{...}\n...else\n"; found "if
     |       | (...)\n...{...}\n...else"
 184 | ERROR | Expected "if (...)\n...{...}\n...else\n"; found "if
     |       | (...)\n...{...}...else"
--------------------------------------------------------------------------------
UPGRADE TO PHP_CODESNIFFER 2.0 TO FIX ERRORS AUTOMATICALLY
--------------------------------------------------------------------------------
```

**Comment by roland-d** [comment]
I don't think the last change is correct. In Joomla 3.4.5 the dropdown looks like this:
finder

Branches only as first option and I think we should keep that.

**Comment by pe7er** [comment]
I removed my previous PR "fixed Select Branch dropdown in Smart Search Content Maps"

**Comment by Kubik-Rubik** [comment]
I have tested this item :white_check_mark: successfully on 26670dd3f5580b7f6426e4ea01942af31ba48dea<br /><br />Works as described. Thank you @pe7er!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8140">issues.joomla.org/joomla-cms/8140</a>.</sub>

**Comment by Kubik-Rubik** [comment]
Works but a small design problem:

2015-11-04 21_32_38-joomla-cms - administration - smart search_ indexed content

This can also be fixed after we've merged this PR!

**Comment by dgt41** [comment]
I have tested this item :white_check_mark: successfully on 26670dd3f5580b7f6426e4ea01942af31ba48dea<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8140">issues.joomla.org/joomla-cms/8140</a>.</sub>

**Comment by dgt41** [comment]
To solve the problem Victor mentioned above just move the filters under the message
like this:
```php
		<?php if (!$this->pluginState['plg_content_finder']->enabled) : ?>
			<div class="alert fade in">
				<button class="close" data-dismiss="alert">×</button>
				<?php echo JText::_('COM_FINDER_INDEX_PLUGIN_CONTENT_NOT_ENABLED'); ?>
			</div>
		<?php endif; ?>
		<?php echo JLayoutHelper::render('joomla.searchtools.default', array('view' => $this)); ?>
```

**Comment by n9iels** [comment]
I spotted that design problem to. Maybe it is an idea to move that message to a `JFactory::getApplication()->enqueueMessage`? There are more component with that problem, where the search filters will be add.

**Comment by zero-24** [comment]
RTC lets merge here and @n9iels or @dgt41 can send a fix for the design problem. Thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8140">issues.joomla.org/joomla-cms/8140</a>.</sub>

**Comment by roland-d** [comment]
The branches dropdown needs to be checked. It is quite different from the current implementation.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8140">issues.joomla.org/joomla-cms/8140</a>.</sub>

**Comment by zero-24** [comment]
Ah sorry overread this comment ;)

**Comment by dgt41** [comment]
Looks good to me:
screenshot 2015-11-04 23 36 38

**Comment by roland-d** [comment]
@dgt41 Compare it to the implementation in J3.4.5, it is quite different. The current implementation uses it's own JHtml class, that is completely skipped now.

**Comment by dgt41** [comment]
ahh yeah, the finder way 😝

**Comment by pe7er** [comment]
I've removed the hardcoded SQL
query="SELECT id AS value, title AS branch FROM #__finder_taxonomy WHERE parent_id = 1"
from /administrator/components/com_finder/models/forms/filter_maps.xml and
added /administrator/components/com_finder/models/fields/branches.php 
to retrieve the branches.

File: administrator/components/com_finder/models/maps.php [hunk 4]
```diff
@@ -263,7 +264,7 @@ protected function populateState($ordering = null, $direction = null)
 		$state = $this->getUserStateFromRequest($this->context . '.filter.state', 'filter_state', '', 'string');
 		$this->setState('filter.state', $state);
 
-		$branch = $this->getUserStateFromRequest($this->context . '.filter.branch', 'filter_branch', '1', 'string');
+		$branch = $this->getUserStateFromRequest($this->context . '.filter.branch', 'filter_branch', '', 'string');
```
Comment (roland-d) [inline administrator/components/com_finder/models/maps.php]: Don't we have to put back the default value here?

**Comment by dgt41** [comment]
@pe7er I am afraid that administrator/components/com_finder/models/fields/branches.php doesn;t cover all the functionality of the old JHtml class https://github.com/joomla/joomla-cms/blob/staging/components/com_finder/helpers/html/filter.php#L208-L394

File: administrator/components/com_finder/models/fields/branches.php [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+	 * Method to get the field options.
+	 *
+	 * @return  array  The field option objects.
+	 *
+	 * @since   3.5
+	 */
+	public function getOptions()
+	{
+		// Build the query.
+		$db = JFactory::getDbo();
```
Comment (roland-d) [inline administrator/components/com_finder/models/fields/branches.php]: @pe7er Just change this complete function to contain:
```
return JHtml::_('finder.mapslist');

**Comment by pe7er** [comment]
Removed the SQL query from fields/branches.php & replaced with return JHtml::_('finder.mapslist');
Thanks for the tip @roland-d !

**Comment by dgt41** [comment]
@pe7er can you also implement this https://github.com/joomla/joomla-cms/pull/8140#issuecomment-153859716
so I can re-test this?

**Comment by pe7er** [comment]
@dgt41 yes, please! 
I think that we should correct the layout issue with the message later because it might need to be corrected at other places as well...

**Comment by dgt41** [comment]
Works fine here, one small UX tho:
screenshot 2015-11-05 01 18 30

**Comment by designbengel** [comment]
Small issue:

Before:
bildschirmfoto 2015-11-05 um 00 22 00

After patch:
bildschirmfoto 2015-11-05 um 00 22 41

**Comment by designbengel** [comment]
I don´t know if this regards this PR but if i don´t have published filters there is a message that "No filters have been created yet.	Create a filter." - that´s not true because there are filters but they are unpublished. 
bildschirmfoto 2015-11-05 um 00 25 04

**Comment by pe7er** [comment]
Thanks for testing @dgt41 & @designbengel !
Those UX issues need to be corrected, but I think that it could be done for beta2 in another PR.

**Comment by designbengel** [comment]
There is a Message that no content has been indexed by filtering unpublished items?

bildschirmfoto 2015-11-05 um 00 27 24

**Comment by roland-d** [comment]
@designbengel I think this is an existing issue, this PR only moves the buttons from the sidebur to the Search Tools bar. Perhaps raise a separate issue for that?

**Comment by designbengel** [comment]
I have tested this item :white_check_mark: successfully on b14cbd40ec4583a94d81a0d7386c7b024314249c<br /><br />Ok, so like @roland-d says it´s regarding just the location-change of filters - so test for this is ok :)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8140">issues.joomla.org/joomla-cms/8140</a>.</sub>

**Comment by infograf768** [comment]
Folks, I can't see
COM_FINDER_MAPS_SELECT_BRANCHE="- Select Branch -"
anywhere.

screen shot 2015-11-27 at 17 43 51

**Comment by dgt41** [comment]
This is normal, you have to be INSIDE a branch for the option to appear (in fact this is a way to go one level back)
screen shot 2015-11-27 at 18 48 56

**Comment by infograf768** [comment]
@pe7er  @roland-d 

I have the feeling some new lang strings were merged here that should'nt have...

**Comment by infograf768** [comment]
@pe7er @roland-d Please test
https://github.com/joomla/joomla-cms/issues/8558

**Pull request closed without merging**